test/token/ERC721/ERC721Holder.behavior.js

```diff
@@ -0,0 +1,25 @@
+const ERC721Holder = artifacts.require('ERC721Holder.sol');
+
+require('chai')
+  .should();
+
+function shouldBehaveLikeERC721Holder (accounts) {
```

Comment:
I'm not sure this name is accurate: the token isn't behaving like an ERC721 Holder, it is instead interacting with one. 

I think we should have a test for the receiver itself, that calls the function and checks the return value, and one for the token, checking that `safeTransferFrom` works properly (i.e. rejects an invalid return value, accepts the correct one). @shrugs is the local ERC721 expert though, so I'll defer to his wisdom.

Reply:
maybe I can rename it to: shouldWorkWithERC721Holder.

Reply:
I checked the tests in some more detail: we're already testing that `safeTransferFrom` in the `describe('via safeTransferFrom')` block, what we need to test is that the ERC721Holder works with a compliant ERC721 token (i.e. it returns the correct value). Does that make sense?

Reply:
It does. But I think that the best way to test that the correct value is returned from the Holder contract is to check that safeTransferFrom succeeds. Just calling onReceived directly and verifying that a magic value is returned is too low level, IMO.

Reply:
My point was that it's not a token test, but a holder test - it belongs in a separate test file :stuck_out_tongue:

Reply:
Yes, that sounds right. I am just wondering if I should keep this style of tests/behavior, even if it's only one simple test.